utils/zerovec/src/zerovec/mod.rs

```diff
@@ -283,29 +285,62 @@ where
     /// ```
     /// use zerovec::ZeroVec;
     ///
-    /// let bytes: &[u8] = &[0xD3, 0x00, 0x19, 0x01, 0xA5, 0x01, 0xCD, 0x01];
-    /// let zv_u16: ZeroVec<u16> = ZeroVec::parse_byte_slice(bytes).expect("infallible");
-    /// let zv_u32: ZeroVec<u32> = zv_u16.try_into_converted()
-    ///     .expect("length is divisible");
+    /// let bytes: &[u8] = &[0x7F, 0xF3, 0x01, 0x00, 0x49, 0xF6, 0x01, 0x00];
+    /// let zv_char: ZeroVec<char> = ZeroVec::parse_byte_slice(bytes)
+    ///     .expect("valid code points");
+    /// let zv_u32: ZeroVec<u32> = zv_char.try_into_converted()
+    ///     .expect("infallible conversion");
     ///
     /// assert!(matches!(zv_u32, ZeroVec::Borrowed(_)));
-    /// assert_eq!(zv_u32.get(0), Some(0x011900D3));
+    /// assert_eq!(zv_u32.get(0), Some(u32::from('🍿')));
     /// ```
     ///
     /// Convert an owned `ZeroVec`:
     ///
     /// ```
     /// use zerovec::ZeroVec;
     ///
-    /// let nums: &[u16] = &[211, 281, 421, 461];
-    /// let zv_u16 = ZeroVec::clone_from_slice(nums);
-    /// let zv_u32: ZeroVec<u32> = zv_u16.try_into_converted()
+    /// let chars: &[char] = &['🍿', '🙉'];
+    /// let zv_char = ZeroVec::clone_from_slice(chars);
+    /// let zv_u32: ZeroVec<u32> = zv_char.try_into_converted()
     ///     .expect("length is divisible");
     ///
     /// assert!(matches!(zv_u32, ZeroVec::Owned(_)));
-    /// assert_eq!(zv_u32.get(0), Some(0x011900D3));
+    /// assert_eq!(zv_u32.get(0), Some(u32::from('🍿')));
+    /// ```
+    ///
+    /// If the types are not the same size, we refuse to convert:
+    ///
+    /// ```should_panic
+    /// use zerovec::ZeroVec;
+    ///
+    /// let bytes: &[u8] = &[0x7F, 0xF3, 0x01, 0x00, 0x49, 0xF6, 0x01, 0x00];
+    /// let zv_char: ZeroVec<char> = ZeroVec::parse_byte_slice(bytes)
+    ///     .expect("valid code points");
+    ///
+    /// // Panics! mem::size_of::<char::ULE> != mem::size_of::<u16::ULE>
+    /// zv_char.try_into_converted::<u16>();
+    /// ```
+    ///
+    /// Instead, convert to bytes and then parse:
+    ///
+    /// ```
+    /// use zerovec::ZeroVec;
+    ///
+    /// let bytes: &[u8] = &[0x7F, 0xF3, 0x01, 0x00, 0x49, 0xF6, 0x01, 0x00];
+    /// let zv_char: ZeroVec<char> = ZeroVec::parse_byte_slice(bytes)
+    ///     .expect("valid code points");
+    /// let zv_u16: ZeroVec<u16> = zv_char.into_bytes().try_into_parsed()
+    ///     .expect("infallible");
+    ///
+    /// assert!(matches!(zv_u16, ZeroVec::Borrowed(_)));
+    /// assert_eq!(zv_u16.get(0), Some(0xF37F));
     /// ```
     pub fn try_into_converted<P: AsULE>(self) -> Result<ZeroVec<'a, P>, <P::ULE as ULE>::Error> {
+        assert_eq!(
+            core::mem::size_of::<<T as AsULE>::ULE>(),
+            core::mem::size_of::<<P as AsULE>::ULE>()
```

Comment:
thought: a thing we could do is that if these sizes are _not_ equal, we can do the allocating variant, which allows you to go back and forth between a `ZeroVec<u8>` (for free in the case of borrowed vecs)

Reply:
Possible, but it would have code size impact.  It's nice to avoid this function having to ship the Vec allocation code.